### ISSUE TITLE
What is the minimum number of classes required by the code?

### ISSUE THREAD

**Issue by rfairhurst** [DrSleep/light-weight-refinenet#47]
I tried running the model with my own dataset using images that had a code of 0 for buildings and 255 (ignore) for everything else and the model reached perfect validation after 1 epoch if I set the number of classes to 1 and 2 epochs if I set the number of classes to 2.  Either way, I assume that is not how the model is supposed to work.

Do I need at least two classes that are not ignored for the model to work?  Would the model work if I assigned a code of 0 for buildings and a code of 1 for everything else in the image?  If that does not work and I have to properly classify a lot of objects in my images that will significantly reduce the number of images I will be able to use with the model, since I only have buildings classified at this point.  I have 4000 images ready to go if I could just classify buildings and make everything else an ignored portion of the image or a single class, but I would only have 100 or less if I have to manually classify real objects in the images other than buildings.  Also if I had a dataset with codes of 0, 1, and 255 (ignore) would the number of classes parameter in the configuration file need to be set to 2 classes or 3 classes?

Additionally, does the model work properly with 32 bit images or does it require 24 bit images like the images in the nyu dataset?

**Comment by rfairhurst** [comment]
I tried running the model with images that had a code of 0 for buildings and 255 (ignore) for everything else and the model reached perfect validation after 1 epoch if I set the number of classes to 1 and 2 epochs if I set the number of classes to 2.  Either way, I assume that is not how the model is supposed to work.

Do I need at least two classes that are not ignored for the model to work?  Would the model work if I assigned a code of 0 for buildings and a code of 1 for everything else in the image?  If that does not work and I have to properly classify a lot of objects in my images that will significantly reduce the number of images I will be able to use with the model, since I only have buildings classified at this point.  I have 4000 images ready to go if I could just classify buildings and make everything else an ignored portion of the image or a single class, but I would only have 100 or less if I have to manually classify real objects in the images other than buildings.  Also if I had a dataset with codes of 0, 1, and 255 (ignore) would the number of classes parameter in the configuration file need to be set to 2 classes or 3 classes?

Also, does the model work properly with 32 bit images or does it require 24 bit images like the images in the nyu dataset?
